app/src/features/Batches/BatchList.tsx

```diff
@@ -82,28 +87,24 @@ const BatchList = (): JSX.Element => {
     }
   );
 
-  const convertDateToTime = (batchStart: string, batchEnd: string) => {
-    const timeBatchStart = new Date(batchStart).getTime();
-    const timeBatchEnd = new Date(batchEnd).getTime();
-    const duration = timeBatchEnd - timeBatchStart;
-    const hours = Math.floor((duration / (1000 * 60 * 60)) % 24);
-    const minutes = Math.floor((duration / (1000 * 60)) % 60);
-    return { hours, minutes };
+  const getDurationString = (duration: number, unit: string) => {
+    if (duration) return `${t(`${unit}`, { count: duration })} `;
+    else return "";
   };
 
   const getBatchDuration = (batch: Batch) => {
     const batchEnd = batch.completed_at ?? batch.canceled_at;
     if (batchEnd) {
-      const { hours, minutes } = convertDateToTime(batch.created_at, batchEnd);
-      if (hours > 0 && minutes > 0) {
-        return `${t("hour", { count: hours })} ${t("minute", {
-          count: minutes,
-        })}`;
-      } else if (hours > 0 && minutes === 0) {
-        return t("hour", { count: hours });
-      } else {
-        return t("minute", { count: minutes });
-      }
+      const endTime = moment.utc(new Date(batchEnd));
+      const duration: Duration = moment.duration(
+        endTime.diff(new Date(batch.created_at))
+      );
+      return `${getDurationString(duration.years(), "year")}
+      ${getDurationString(duration.months(), "month")}
+      ${getDurationString(duration.days(), "day")}
+      ${getDurationString(duration.hours(), "hour")}
+      ${getDurationString(duration.minutes(), "minute")}
```

Comment:
so if the batches lasts 1 minute it displays:
`0 year 0 month 0 day 0 hour 1 minute` ?
please use https://momentjs.com/docs/#/displaying/format/ this is the whole point of using a lib for this!

Reply:
https://stackoverflow.com/questions/13262621/how-to-use-format-on-a-moment-js-duration

Reply:
No it's not !
![image](https://user-images.githubusercontent.com/47388675/133648120-6f415743-f226-4fb1-94c5-bce3c158046c.png)

There is no way to display durations without putting 0 everywhere with moment.js. Or maybe I didn't check that well...